ddtrace/_monkey.py

```diff
@@ -201,6 +200,11 @@ def patch_all(**patch_modules):
 
     patch(raise_errors=False, **modules)
     patch_iast(**IAST_PATCH)
+    if config._appsec_enabled:
+        patched_modules = patch_appsec_subprocess_executions()
+        with _LOCK:
+            for patched in patched_modules:
+                _PATCHED_MODULES.add(patched)
```

Comment:
Suggestion: Right now subprocess executions are only patched if `patch_all()` is used. Can we use a pattern similar to the other integrations?  Ex: define a "subprocess" integration in `PATCH_MODULES` and then add the following on line 252 (in _monket.patch(...)).

```
    if module == "subprocess" and config._appsec_enabled:
        from .appsec._patch_subprocess_executions import _patch
        _patch()
        continue
```

This approach allows customers to opt in or out of subprocess tracing using ``DD_TRACE_SUBPROCESS_ENABLED`` however this is not a typical integration so this code block feels a bit out of place.

An alternative is to define integrations for both os and subprocess.

Again this suggestion is non-blocking the current approach works

Reply:
I still didn't have a clear answer on this (probably the people I contacted didn't know for sure either) but it looks like this will be extended in the future (just like we'll be extending login events soon) to be able to block subprocess executions using RCM so I kept it in the ASM product directory. Regarding the options like the env var you suggested, the current RFC indicated that the feature should be enabled or disabled totally depending only if ASM is enabled.